### PR TITLE
601 - IdsDropdown Remaining Features

### LINKED ISSUE #601: IdsDropdown: Remaining Features/Issues
**Description**
Lists the Remaining Bugs/Features for a 100% Component

**Major**
- [x] Bug No initial value and you cant select see https://github.com/infor-design/enterprise-wc/issues/600
- [x] Groups like https://main-enterprise.demo.design.infor.com/components/dropdown/example-groups.html
- [x] Placeholder / Blank Item
- [x] #759
- [x] #758
- [x] Add all size examples like https://main-enterprise.demo.design.infor.com/components/dropdown/example-widths.html (width and height) #674
- [x] Check themes with old version (dark theme is incorrect) see https://github.com/infor-design/enterprise-wc/issues/600

**Minor**

- [x] Bug when opening it in the slight wrong position
- [x] Test duplicate values/labels
- [x] Text xss like https://main-enterprise.demo.design.infor.com/components/dropdown/test-xss.html
- [x] Add events or funnel events from the popup when it opens and closes
- [x] Make sure open issues get in when fixed https://github.com/infor-design/enterprise/issues?q=is%3Aissue+is%3Aopen+dropdown

**Maybe in the Future**
- [ ] Virtual Scrolling

### PR DESCRIPTION
**Pull request by andriivarhanov**
**Explain the details for making this change. What existing problem does the pull request solve?**
This PR includes `ids-dropdown` remaining features/fixes:
- groups
- placeholder
- duplicated values example
- xss test 
- clearable with icons fix
- typeahead with icons fix
- other small adjustments and fixes

**Related github/jira issue (required)**:
Closes #601

**Steps necessary to review your pull request (required)**:
- pull and build
- go to http://localhost:4300/ids-dropdown/groups.html
- check grouping functionality
- go to http://localhost:4300/ids-dropdown/sandbox.html
- check placeholder/xss/duplicated examples
- go to http://localhost:4300/ids-dropdown/typeahead.html
- check "Typeahead with icons" example
- go to http://localhost:4300/ids-dropdown/clearable.html
- check "Clearable with Icons" example

**Included in this Pull Request**:
- [x] An e2e or functional test for the bug or feature.
~- [ ] A note to the change log.~

### CODE REVIEW

File: src/components/ids-dropdown/demos/sandbox.html [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+            <ids-list-box-option id="wa" value="wa">Washington</ids-list-box-option>
+            <ids-list-box-option id="wv" value="wv">West Virginia</ids-list-box-option>
+            <ids-list-box-option id="wi" value="wi">Wisconsin</ids-list-box-option>
+            <ids-list-box-option id="wy" value="wy">Wyoming</ids-list-box-option>
+          </ids-list-box>
+        </ids-dropdown>
+
+        <ids-dropdown clearable="true" label="Xss Test">
+          <ids-list-box>
+            <ids-list-box-option value="quote">"quote"</ids-list-box-option>
```
Comment (EdwardCoyle) [inline src/components/ids-dropdown/demos/sandbox.html:64]: There is something strange about how these get displayed in the real Dropdown -- All the words in the actual list item appear capitalized, but the contents here aren't:

Screen Shot 2022-08-22 at 10 17 41 AM

**Review by EdwardCoyle — APPROVED** [review]
@andriivarhanov all looks good here -- just had a comment about this, not sure if this is related to this PR or not, but calling out just in case

**Review by tmcconechy — APPROVED** [review]
Looks good to me too. No issues found.

**Comment by andriivarhanov** [comment]
@EdwardCoyle there is capitalized CSS setting for each dropdown option, but not for the input. I'm not sure if it's an issue or not, we just didn't have examples with lowercase text options to notice that. Of course, I can remove it, just let me know

**Pull request merged**